lib/reportportal.rb

```diff
@@ -170,25 +142,13 @@ def close_child_items(parent_id)
 
     private
 
-    def create_resource(url)
-      props = { :headers => {:Authorization => "Bearer #{Settings.instance.uuid}"}}
-      verify_ssl = Settings.instance.disable_ssl_verification
-      props[:verify_ssl] = !verify_ssl unless verify_ssl.nil?
-      RestClient::Resource.new url, props
-    end
-
-    def do_request(url)
-      resource = create_resource(url)
-      tries = 3
-      begin
-        yield resource
-      rescue
-        p "Request to #{url} produced an exception: #{$!.class}: #{$!}"
-        $!.backtrace.each { |l| p l }
-        retry unless (tries -= 1).zero?
```

Comment:
Ok, I added retries. They could be useful for us in case of e.g. intermittent connection failures. I didn't consider them initially. But 4xx/5xx errors would likely be reproducible even after more attempts. I added retries for them as well.